src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Filters/IdentityTelemetryFilter.cs

```diff
@@ -0,0 +1,131 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Security.Claims;
+
+using AltinnCore.Authentication.Constants;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Http;
+
+namespace Altinn.Platform.Authorization.Filters
+{
+    /// <summary>
+    /// Filter to enrich request telemetry with identity information
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public class IdentityTelemetryFilter : ITelemetryProcessor
+    {
+        private ITelemetryProcessor Next { get; set; }
+
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IdentityTelemetryFilter"/> class.
+        /// </summary>
+        public IdentityTelemetryFilter(ITelemetryProcessor next, IHttpContextAccessor httpContextAccessor)
+        {
+            Next = next;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        /// <inheritdoc/>
+        public void Process(ITelemetry item)
+        {
+            RequestTelemetry request = item as RequestTelemetry;
+
+            if (request != null && request.Url.ToString().Contains("authorization/api/"))
+            {
+                HttpContext ctx = _httpContextAccessor.HttpContext;
+
+                if (ctx?.User != null)
+                {
+                    int? orgNumber = GetOrgNumber(ctx.User);
+                    int? userId = GetUserIdAsInt(ctx.User);
+                    int? partyId = GetPartyIdAsInt(ctx.User);
+                    int authLevel = GetAuthenticationLevel(ctx.User);
+
+                    string idString = $"partyId:{partyId};userId:{userId};orgNumber:{orgNumber};authLevel:{authLevel}";
+
+                    item.Context.User.AuthenticatedUserId = GetUserOrOrgId(ctx.User);
+                    item.Context.User.Id = idString;
```

Comment:
Consider the use of properties here as well ref. comment on https://github.com/Altinn/altinn-studio/pull/6687